t/unit/backends/test_cache.py

```diff
@@ -20,6 +21,22 @@ def __init__(self, data):
         self.data = data
 
 
+class test_backend_by_url:
+    """We can't test these in t/unit/app/test_backends, because of importing in
```

Comment:
```
$ ack -l def\ test_backend_by_url
t/unit/backends/test_mongodb.py
t/unit/backends/test_dynamodb.py
t/unit/backends/test_elasticsearch.py
t/unit/backends/test_couchdb.py
t/unit/backends/test_riak.py
t/unit/backends/test_couchbase.py
```

Plenty of precedent for this. I just added cache, database, and mongo tests to fill out the suite. DyanamoDB, Elasticsearch, Couchdb, and Riak were already there.

MongoDB doesn't get tested with the current Travis setup (probably an oversight?). Don't know about the rest. I just copied the existing paradigm for tests that require non-Travis.yaml libraries to be present. Personally, I think this is the right factoring. Adding all of these tests to test_backends.py will clutter it up with a bunch of skip.unless_module decorators, and since it doesn't actually import the module into the current scope, a local import would be needed for each as well.